### PR TITLE
Update k8s repository links and fix docs links

### PR DESCRIPTION
**Pull request by soltysh**
Since k8s moved to its own group on gh, I've reworked the links, fixing those pointing at docs at the same time.

### CODE REVIEW

File: hack/conversion/conversion.go [hunk 1]
```diff
@@ -55,7 +55,7 @@ func main() {
 	generator.OverwritePackage(*version, "")
 	generator.OverwritePackage("api", "newer")
 	for _, knownType := range kapi.Scheme.KnownTypes(*version) {
-		if strings.Contains(knownType.PkgPath(), "GoogleCloudPlatform/kubernetes") {
+		if strings.Contains(knownType.PkgPath(), "k8s.io/kubernetes") {
```
Comment (soltysh) [inline hack/conversion/conversion.go]: @smarterclayton that's my only doubt, is that leftover from @deads2k #4121 ?

Comment (deads2k) [inline hack/conversion/conversion.go]: > @smarterclayton that's my only doubt, is that leftover from @deads2k #4121 ?

Is this used by anything.  We use `cmd/genconversion/conversion.go` for generated conversions.

**Comment by soltysh** [comment]
@liggitt the travis failure might be something you'd be interested in, as it fail terribly on TestOAuthLDAP, can you have a look at it? https://travis-ci.org/openshift/origin/jobs/75422510

File: hack/conversion/conversion.go (hunk 1 above)
Comment (soltysh) [inline hack/conversion/conversion.go]: Yeah, it looks like a leftover. I'll remove it.

Comment (soltysh) [inline hack/conversion/conversion.go]: Especially that it looks very similarly to `cmd/genconversion/conversion.go` but with less functionality.

File: hack/conversion/conversion.go [hunk 2]
```diff
@@ -1,71 +0,0 @@
-package main
```
Comment (rhcarvalho) [inline hack/conversion/conversion.go]: @soltysh why are you deleting this file?

**Comment by mfojtik** [comment]
@soltysh why are you removing conversion.go?

File: hack/conversion/conversion.go (hunk 2 above)
Comment (mfojtik) [inline hack/conversion/conversion.go]: @rhcarvalho too late!

**Comment by mfojtik** [comment]
LGTM (just make the removal of conversion.go a separate commit with reasonable commit message providing information about why it was removed)

**Comment by soltysh** [comment]
@mfojtik @rhcarvalho just fyi, that was removed because we are using `cmd/genconversion/conversion.go` instead for quite some time. Apparently it wasn't removed any sooner. Moved that removal to separate commit, per @mfojtik's request. Will wait for tests to finish...

File: hack/conversion/conversion.go (hunk 2 above)
Comment (rhcarvalho) [inline hack/conversion/conversion.go]: I commented earlier than you :stuck_out_tongue:

**Comment by soltysh** [comment]
[merge]

**Pull request merged**